rust/gui-client/src-tauri/src/client/tunnel-wrapper/in_proc.rs

```diff
@@ -0,0 +1,116 @@
+//! In-process wrapper for connlib
+//!
+//! This is used so that Windows can keep connlib in the GUI process a little longer,
+//! until the Linux process splitting is settled. Once both platforms are split,
+//! this should be deleted.
+//!
+//! With this module, the main GUI module should have no direct dependence on connlib.
+//! And some things in here will live in the tunnel process after the IPC split.
+//! (It's okay to depend on trivial things like `BUNDLE_ID`)
+
+use anyhow::{Context, Result};
+use arc_swap::ArcSwap;
+use connlib_client_shared::{file_logger, ResourceDescription, Sockets};
+use connlib_shared::{keypair, LoginUrl};
+use secrecy::SecretString;
+use std::{
+    net::{IpAddr, Ipv4Addr, Ipv6Addr},
+    sync::Arc,
+    time::Duration,
+};
+use tokio::sync::Notify;
+
+use super::ControllerRequest;
+use super::CtlrTx;
+
+/// The Windows client doesn't use platform APIs to detect network connectivity changes,
+/// so we rely on connlib to do so. We have valid use cases for headless Windows clients
+/// (IoT devices, point-of-sale devices, etc), so try to reconnect for 30 days if there's
+/// been a partition.
+/// (Update: This may no longer be needed, we do detect network changes now)
```

Comment:
Worth fixing in a separate before merging this?

Reply:
I removed the part about network connectivity changes, but I'm not sure how the partition time works so I didn't open any other changes for it yet.

I guess partition time means the length of time before connlib gives up on reconnecting?

So before, Windows Client doesn't watch for network changes, connlib just keeps trying to reconnect no matter what, so it needs a very long max partition time, otherwise it would give up in a situation where a human won't be around to reboot it

And after, Windows Client does listen for network changes, so we tell connlib to immediately reconnect if we regain Internet. (actually I'm not sure what it does for loss of Internet, I'd have to re-read the code)

But is there any advantage to shortening the max partition time? Could we just keep it long?